examples/e2e/pipelines/batch_inference.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from config import MetaConfig, PipelinesConfig
+from steps import (
+    data_loader,
+    drift_na_count,
+    inference_data_preprocessor,
+    inference_predict,
+    notify_on_failure,
+    notify_on_success,
+)
+from utils.artifacts import find_artifact_id
+
+from zenml import pipeline
+from zenml.integrations.evidently.metrics import EvidentlyMetricConfig
+from zenml.integrations.evidently.steps import evidently_report_step
+from zenml.logger import get_logger
+from zenml.steps.external_artifact import ExternalArtifact
+
+logger = get_logger(__name__)
+
+
+@pipeline(
+    settings={"docker": PipelinesConfig.docker_settings},
+    on_success=notify_on_success,
+    on_failure=notify_on_failure,
+)
+def e2e_example_batch_inference():
+    """
+    Model batch inference pipeline.
+
+    This is a pipeline that loads the inference data, processes
+    it, analyze for data drift and run inference.
+    """
+    ### ADD YOUR OWN CODE HERE - THIS IS JUST AN EXAMPLE ###
+    # Link all the steps together by calling them and passing the output
+    # of one step as the input of the next step.
+    ########## ETL stage  ##########
+    dataset_inf = data_loader(
+        n_samples=10_000,
+        drop_target=True,
+    )
+    preprocess_pipeline_id = find_artifact_id(
+        pipeline_name=MetaConfig.pipeline_name_training,
+        artifact_name="preprocess_pipeline",
+    )
+    dataset_inf = inference_data_preprocessor(
+        dataset_inf=dataset_inf,
+        preprocess_pipeline=ExternalArtifact(id=preprocess_pipeline_id),
```

Comment:
Another interesting use I haven't seen before. Very cool if this works since it doesn't redundantly save the artifact again. If not, you could try wrapping `find_artifact_id` in a `step` instead which does `artifact.load()` to load and returns the artifact again (but, as I said, that would be less efficient since the artifact would be redundantly saved again then).

Reply:
Yes, it works. I also didn't want to go with load and passing cause of the double saving issue.
I had to use the standard function to evaluate `id` on pipeline compose, cause otherwise it would fail.
I was also thinking to put this helper function somewhere in the standard, cause it seems to be a rather generic case for 2+  pipelines per product.

Reply:
Ok, I went high stakes and modified the `ExternalArtifact` class to fit the purpose.
Can you have a close look [here](https://github.com/zenml-io/zenml/pull/1710/commits/fe0111892e2a53ce4c27dcd78ef0a68bcfe61dd7) please @schustmi @fa9r @bcdurak